src/doc/src/reference/rust-version.md

```diff
@@ -0,0 +1,154 @@
+# Rust Version
+
+The `rust-version` field is an optional key that tells cargo what version of the
+Rust toolchain you support for your package.
+
+```toml
+[package]
+# ...
+rust-version = "1.56"
+```
+
+The Rust version must be a bare version number with at least one component; it
+cannot include semver operators or pre-release identifiers. Compiler pre-release
+identifiers such as -nightly will be ignored while checking the Rust version.
+
+> **MSRV:** Respected as of 1.56
+
+## Uses
+
+**Diagnostics:**
+
+When your package is compiled on an unsupported toolchain,
+Cargo will provide clearer diagnostics than reporting invalid syntax or missing functionality in `std`.
+This affects all build-targets in the package, including binaries, examples, test suites,
+benchmarks, etc.
+
+**Development aid:**
+
+`cargo add` will auto-select the dependency's version requirement to be low enough to be compatible with your `rust-version`.
```

Comment:
```suggestion
`cargo add` will auto-select the dependency's version requirement to be the latest version compatible with your `rust-version`, which may not be the latest version of the dependency. `cargo add` will warn if it selects an older version, and tell you what MSRV would be required to use the latest version.
```

Reply:
For the second half of that, I feel like that is clear from the first half and would rather not get into the specifics of how we behave.

Reply:
I think "which may not be the latest version of the dependency" naturally follows but is still helpful reinforcement to help people who aren't familiar with the problem domain wrap their heads around it.

For the latter half, the exact details aren't load-bearing. The main things I'm trying to convey are:
- `cargo add` doesn't do this *silently*
- If you'd rather use the latest version, here's how you can know what MSRV you'd need to use to do that: `cargo add` will tell you

Reply:
Further tweaked it.

Reply:
LGTM, thanks!